crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -303,14 +307,33 @@ impl<'db> CallBindingError<'db> {
     ) {
         match self {
             Self::InvalidArgumentType {
+                callable_ty,
                 parameter,
                 argument_index,
                 expected_ty,
                 provided_ty,
             } => {
+                let mut messages = vec![];
+                if let Some(func_lit) = callable_ty.into_function_literal() {
+                    let func_lit_scope = func_lit.body_scope(context.db());
+                    let mut span = Span::from(func_lit_scope.file(context.db()));
+                    let node = func_lit_scope.node(context.db());
+                    if let Some(func_def) = node.as_function() {
+                        let range = argument_index
+                            .and_then(|i| func_def.parameters.iter().nth(i))
```

Comment:
In the presence of synthetic arguments like `self`, this matching of argument indices to parameter indices is not always correct, I'm afraid. Consider this example:
```py
class C:
    def __call__(self, x: int) -> int:
        return 1


c = C()
c("wrong")
```

which currently results in a wrong highlighting of the `self` parameter:

![image](https://github.com/user-attachments/assets/cf821f4f-82d4-4cea-80af-fa9645d73f19)

It looks like this can be fixed by taking the information from `parameter.index` instead?

```suggestion
                        let range = func_def
                            .parameters
                            .iter()
                            .nth(parameter.index)
```

This would also lead to more correct ranges for functions with variadic arguments, where we currently highlight *all parameters* if the argument index is (seemingly) "out of bounds":

```py
def f(x: int, *y: str):
    pass

f(1, "foo", None)
```

![image](https://github.com/user-attachments/assets/a04cafaa-ec3d-42fd-b110-0ddad6b8671c)

Reply:
Nice catches. I added a regression test for the first case. And I think the second case is covered by an existing test. (The span still points to all parameters, but now it doesn't include the parentheses.)